techmatters/hrm
# CHI-2359: case status transition task

## Description

* Adds new scheduled task that updates the status of cases based on rules specified in SSM parameters
* The rules are very simple, you can specify a starting status, the minimum amount of time something has to be in that status to be valid for transition and the target status
* You also specify a description, this will be used in the statusUpdatedBy field and shown in aselo in the status line on the EditCaseSummary page
* Fix the mock proxy lifecycle management - should hopefully stop weird, hard to debug service test errors popping up in the future

TODO: 

* Update flex to show the statusUpdatedBy correctly when updated by a rule
* Update infrastructure config to deploy the new task


### Checklist

### Related Issues
CHI-2359

### Verification steps

* Infrastructure config needs to be updated & the task deployed
* Add a rule in the SSM parameters for the account you want to test with a short duration
* Create a task & put it in the valid state to be transitioned
* Wait for the time specified in the rule to pass and check the status, it should have changed to the targetStatus specified in the rule
* Also (once flex is updated & deployed), check that the edit case summary page shows the rule description as the updater
